torchtext/data/metrics.py

```diff
@@ -0,0 +1,93 @@
+import math
+import collections
+from torchtext.data.utils import ngrams_iterator
+
+
+def bleu_score(candidate_corpus, references_corpus, max_n=4, weights=[0.25] * 4):
+    """Computes the BLEU score between a candidate translation corpus and a references
+    translation corpus. Based on https://www.aclweb.org/anthology/P02-1040.pdf
+
+    Arguments:
+        candidate_corpus: a list of candidate translations. Each translation is a list of
+            tokens
+        references_corpus: a list of lists of reference translations. Each translation
+            is a list of tokens
+        max_n: the maximum n-gram we want to use. E.g. if max_n=3, we will use unigrams,
+            bigrams and trigrams
+        weights: weights used for each n-gram category (uniform by default)
+
+    Examples:
+        >>> candidate_corpus = [['I', 'ate', 'the', 'apple'], ['I', 'did']]
+        >>> references_corpus = [[['I', 'ate', 'it'], ['I', 'ate', 'apples']],
+                [['I', 'did']]]
+        >>> bleu_score(candidate_corpus, references_corpus)
+        >>> 0.7598356856515925
+    """
+
+    def _compute_ngram_counter(tokens, max_n):
```

Comment:
Given on your (very clear) exposition of the different BLUE alternatives this sounds like something that'll become generally useful for the other metrics as well.

So it could be at the top-level (but still marked as private). Of course we can cross that bridge once we get there.

Reply:
I second that.

Reply:
That's true, I'll change that right now it's an easy change